### DISCUSSION TITLE
MySQL / Getting Last Inserted ID for Master / Detail

### DISCUSSION THREAD

**Discussion by poirazis** [Budibase/budibase#5848] [Help]
MySQL can do a `select last_insert_id();` to return the last inserted id ( per client connection )

Setting up a query inside BB to execute "select last_insert_id();" always returns 0.

Is this a known limitation ? a bug ?

**Comment by melohagan** [comment]
Hey @poirazis 

I don't think this is necessarily a Budibase issue: https://stackoverflow.com/questions/13880267/mysql-last-insert-id-returns-0

I was getting 0 returned back directly in MySQL Workbench. When I tried the suggestion with the insert followed by the select I was able to get a returned value.

There is however a limitation in Budibase in that you can only run one SQL statement in a query - not two.

**Comment by poirazis** [comment]
Thank you @melohagan !
